app/src/integrationTests/IntegrationTest.ts

```diff
@@ -61,62 +54,23 @@ export abstract class IntegrationTestSuite {
                     .catch(e => done(e));
             });
 
-            beforeEach((done: DoneCallback) => {
+            beforeEach(async () => {
                 (global as any).fetch = require('node-fetch');
                 singletonVariableCache.clearAll();
-                // Note that this will always trigger an authActions.logIn, which will result in all three login
-                // stores recording the user to some extent
 
                 this.store = this.createStore();
-                this.store.dispatch(authActionCreators.logIn(this.user, "password"));
-                let unsubscribe = this.store.subscribe(handleChange);
-                let that = this;
-
-                function handleChange() {
-                    const token = that.store.getState().auth.bearerToken;
+                const response = await (new AuthService(this.store.dispatch, this.store.getState))
+                    .logIn(this.user, "password");
 
-                    unsubscribe();
-                    done();
-                }
+                initialAuthState.bearerToken = response.access_token;
+                await this.store.dispatch(authActionCreators.loadAuthenticatedUser());
             });
 
             this.addTestsToMocha();
         });
     }
 }
 
-export async function firstDownloadPromise(rendered: ReactWrapper) {
```

Comment:
Just noticed a bunch of helpers methods that were used in the reportle tests are now not in use.